src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -859,10 +863,10 @@ protected EC2SpotSlave newSpotSlave(SpotInstanceRequest sir, String name) throws
     /**
      * Get a KeyPair from the configured information for the slave template
      */
-    private KeyPair getKeyPair(AmazonEC2 ec2) throws IOException, AmazonClientException {
+    private KeyPair getKeyPair(AmazonEC2 ec2) throws IOException {
         KeyPair keyPair = parent.getPrivateKey().find(ec2);
         if (keyPair == null) {
-            throw new AmazonClientException("No matching keypair found on EC2. Is the EC2 private key a valid one?");
+            LOGGER.log(Level.WARNING, "No matching keypair found on EC2. Is the EC2 private key a valid one?");
         }
```

Comment:
I would not have this be a WARNING, but INFO and have the text indicate the circumstances where the key is not required, as this could be part of normal operation.